modules/internal/String.chpl

```diff
@@ -2215,14 +2215,20 @@ module String {
     return ret;
   }
 
+  private proc stringValDeprecated() {
+    compilerWarning("'+' is deprecated for string/value pairs; please cast the value to string instead");
+  }
```

Comment:
While this wording is reasonable, please consider changing it slightly to something like this: `+ is deprecated for string/non-string pairs; please add an explicit cast from the non-string value to string`.

Reply:
How about `'+' between strings and non-strings is deprecated; consider explicitly casting the non-string argument to a string`?

Reply:
Works for me!